azext_edge/edge/providers/check/dataflow.py

```diff
@@ -0,0 +1,906 @@
+                ("Mode", mode),
+            ]:
+                # TODO - validate profile ref
+                check_manager.add_display(
+                    target_name=target,
+                    namespace=namespace,
+                    display=Padding(f"{label}: {val}", (0, 0, 0, padding + 4)),
+                )
+            operations = spec.get("operations", [])
+            if operations and detail_level > ResourceOutputDetailLevel.summary.value:
+                check_manager.add_display(
+                    target_name=target,
+                    namespace=namespace,
+                    display=Padding("Operations:", (0, 0, 0, padding + 4)),
+                )
+            # TODO - error if no operations?
+            processor_dict = {
+                DataflowOperationType.source.value: _process_dataflow_sourcesettings,
+                DataflowOperationType.builtin_transformation.value: _process_dataflow_transformationsettings,
+                DataflowOperationType.destination.value: _process_dataflow_destinationsettings,
+            }
+
+            for operation in operations:
+                # TODO - group by type
+                # TODO - "Dataflow must have 3 operations max"
+                # TODO - "Dataflow must have a local MQ source or a local MQ target."
+                op_type = operation.get("operationType")
+                if op_type and op_type.lower() not in processor_dict:
+                    raise CLIInternalError(f"Invalid operation type {op_type}")
+                processor_dict[op_type.lower()](
+                    check_manager=check_manager,
+                    target=target,
+                    namespace=namespace,
+                    resource=operation,
+                    detail_level=detail_level,
+                    padding=padding + 8,
+                )
+
+            check_manager.add_target_eval(
+                target_name=target,
+                namespace=namespace,
+                status=CheckTaskStatus.success.value,
+                resource_name=dataflow_name,
+                resource_kind=DataflowResourceKinds.DATAFLOW.value,
+            )
+    return check_manager.as_dict(as_list=as_list)
+
+
+def evaluate_dataflow_endpoints(
+    as_list: bool = False,
+    detail_level: int = ResourceOutputDetailLevel.summary.value,
+    resource_name: str = None,
+):
+    check_manager = CheckManager(
+        check_name=dataflow_endpoint_check_name,
+        check_desc=dataflow_endpoint_check_desc,
+    )
+    all_endpoints = get_resources_by_name(
+        api_info=DATAFLOW_API_V1B1,
+        kind=DataflowResourceKinds.DATAFLOWENDPOINT,
+        resource_name=resource_name,
+    )
+    target = dataflow_endpoint_target
+    padding = 8
+    if not all_endpoints:
+        no_endpoints_text = "No Dataflow Endpoints detected in any namespace."
+        check_manager.add_target(target_name=target)
+        check_manager.add_target_eval(
```

Comment:
wondering what would be the --as-object look like for this target? I'm thinking if we should add a corresponding condition when we add an evaluation, otherwise it will just be `conditions=null`🤔